core/ui/PageControls/export-twcore.tid

```diff
@@ -0,0 +1,19 @@
+title: $:/core/ui/Buttons/export-twcore
+tags: $:/tags/PageControls
+caption: {{$:/core/images/star-filled}} {{$:/language/Buttons/ExportTWCore/Caption}}
+description: {{$:/language/Buttons/ExportTWCore/Hint}}
+
+\whitespace trim
+\define jsFileName() twcore-$(version)$.js
+
+<$set name=isButtonOff filter="[[$:/boot/boot.js]is[missing]then[yes]else[no]]">
+<$button tooltip={{$:/language/Buttons/ExportTWCore/Hint}} aria-label={{$:/language/Buttons/ExportTWCore/Caption}} class=<<tv-config-toolbar-class>> disabled=<<isButtonOff>>>
```

Comment:
It introduces quite a lot of complexity to handle the disabled button status, is it really necessary?

Reply:
Hi @Jermolene 

```
\define jsFileName() tiddlywikicore-$(version)$.js

<$set name=isButtonOff filter="[[$:/boot/boot.js]is[missing]then[yes]else[no]]">
<$button tooltip={{$:/language/Buttons/ExportTWCore/Hint}} aria-label={{$:/language/Buttons/ExportTWCore/Caption}} class=<<tv-config-toolbar-class>> disabled=<<isButtonOff>>>
```
> It introduces quite a lot of complexity to handle the disabled button status, is it really necessary?

I don't quite follow your concerns here. What complexity are you referring to? If it's whether the button needs to be disabled in certain situations, then I would argue yes--if the wiki's core has been externalized; otherwise, the users may not realize the exported core from an externalized-core wiki is not usable.

Reply:
If we do need to disable the button we should use a filtered transclusion instead of a set widget:

`<$button tooltip={{$:/language/Buttons/ExportTWCore/Hint}} aria-label={{$:/language/Buttons/ExportTWCore/Caption}} class=<<tv-config-toolbar-class>> disabled={{{[[$:/boot/boot.js]is[missing]then[yes]else[no]]}}}>`

Reply:
>  Sorry, I don't quite follow your concerns here. What complexity are you referring to? If it's whether the button needs to be disabled in certain situations, then I would argue yes--if the wiki's core has been externalized; otherwise, the users may not realize the exported core from an externalized-core wiki is not usable.

The core doesn't currently use the disabled attribute on toolbar buttons, which is why you had to add a CSS class to help style it. Part of the reason that we don't use them is because testing shows that users can find it hard to distinguish disabled buttons from ordinary buttons, and that the behaviour of a disabled button is confusing because it feels like the mouse isn't working. If there's an error condition that prevents a button from being used then I think it's better to display an error message when the button is clicked.

The complexity I was referring to was in the implementation details, but perhaps a bigger concern is usability.

Reply:
> ... we should use a filtered transclusion instead of a set widget:

Thanks @saqimtiaz. I'll keep that mind for the future.